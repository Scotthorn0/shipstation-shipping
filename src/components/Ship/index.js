import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Segment, Modal, Button } from 'semantic-ui-react';
import VerifyAddress from '../../containers/VerifyAddress';
import ConfirmAddress from '../../containers/ConfirmAddress';
import GetRates from '../../containers/GetRates';
import './styles.css';
import { NEXT } from '../../constants';

class Ship extends Component {
  static propTypes = {
    requestAddressVerification: PropTypes.func,
  }

  state = {
    step: 0,
    modalOpen: false,
  }

  moveStep = (direction) => {
    let { step } = this.state;
    const nextStep = direction === NEXT ? step += 1 : step -= 1;
    this.setState({ step: nextStep });
  }

  toggleModal = () => {
    this.setState({ modalOpen: !this.state.modalOpen });
  }

  renderStep() {
    const { step } = this.state;
    const steps = [
      VerifyAddress,
      ConfirmAddress,
      GetRates,
    ];
    return steps[step];
  }

  render() {
    const CurrentStep = this.renderStep();
    const { modalOpen } = this.state;

    return (
      <Segment basic>
        <Modal
          closeOnDimmerClick={false}
          closeIcon
          open={modalOpen}
          onClose={this.toggleModal}
          size="small"
          trigger={
            <Button onClick={this.toggleModal}>Order</Button>
          }
        >
          <Modal.Content>
            <CurrentStep moveStep={this.moveStep} />
          </Modal.Content>
        </Modal>
      </Segment>
    );
  }
}

export default Ship;
