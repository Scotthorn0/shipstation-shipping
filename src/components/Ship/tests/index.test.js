import React from 'react';
import { shallow } from 'enzyme';
import Ship from '../';
import { NEXT, BACK } from '../../../constants';

describe('Ship Component', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(<Ship />);
    instance = wrapper.instance();
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('moveStep', () => {
    test(`should increase the step tracker if the direction is ${NEXT}`, () => {
      expect(wrapper.state('step')).toBe(0);
      instance.moveStep(NEXT);
      expect(wrapper.state('step')).toBe(1);
    });
    test(`should decrease the step tracker if the direction is ${BACK}`, () => {
      wrapper.setState({ step: 2 });
      instance.moveStep(BACK);
      expect(wrapper.state('step')).toBe(1);
    });
  });

  describe('toggleModal', () => {
    test('toggle the modal on and off', () => {
      expect(wrapper.state('modalOpen')).toBe.false;
      instance.toggleModal();
      expect(wrapper.state('modalOpen')).toBe.true;
      instance.toggleModal();
      expect(wrapper.state('modalOpen')).toBe.false;
    });
  });
});
