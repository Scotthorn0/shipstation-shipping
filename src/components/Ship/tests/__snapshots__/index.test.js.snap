// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ship Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ship />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "basic": true,
      "children": <Modal
        closeIcon={true}
        closeOnDimmerClick={false}
        closeOnDocumentClick={false}
        dimmer={true}
        eventPool="Modal"
        onClose={[Function]}
        open={false}
        size="small"
        trigger={
                <Button
                        as="button"
                        onClick={[Function]}
                >
                        Order
                </Button>
        }
>
        <ModalContent>
                <Connect(VerifyAddress)
                        moveStep={[Function]}
                />
        </ModalContent>
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ModalContent>
          <Connect(VerifyAddress)
                    moveStep={[Function]}
          />
</ModalContent>,
        "closeIcon": true,
        "closeOnDimmerClick": false,
        "closeOnDocumentClick": false,
        "dimmer": true,
        "eventPool": "Modal",
        "onClose": [Function],
        "open": false,
        "size": "small",
        "trigger": <Button
          as="button"
          onClick={[Function]}
>
          Order
</Button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Connect(VerifyAddress)
            moveStep={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "moveStep": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "basic": true,
        "children": <Modal
          closeIcon={true}
          closeOnDimmerClick={false}
          closeOnDocumentClick={false}
          dimmer={true}
          eventPool="Modal"
          onClose={[Function]}
          open={false}
          size="small"
          trigger={
                    <Button
                              as="button"
                              onClick={[Function]}
                    >
                              Order
                    </Button>
          }
>
          <ModalContent>
                    <Connect(VerifyAddress)
                              moveStep={[Function]}
                    />
          </ModalContent>
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ModalContent>
            <Connect(VerifyAddress)
                        moveStep={[Function]}
            />
</ModalContent>,
          "closeIcon": true,
          "closeOnDimmerClick": false,
          "closeOnDocumentClick": false,
          "dimmer": true,
          "eventPool": "Modal",
          "onClose": [Function],
          "open": false,
          "size": "small",
          "trigger": <Button
            as="button"
            onClick={[Function]}
>
            Order
</Button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Connect(VerifyAddress)
              moveStep={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "moveStep": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
